feature/markNextVersionTask-creates-next-version-with-incrementer

jplucinski opened the pull request:
* markNextVersionTask can create next version with incrementer

Commit:
feature/markNextVersionTask-creates-next-version-with-incrementer

* markNextVersionTask can create next version with incrementer

toniedzwiedz commented on docs/configuration/next_version.rst:
```diff
@@ -10,19 +10,21 @@ version number and last tag is next version marker, version is treated as SNAPSH
 
-To create next version marker use ``markNextVersion`` task along with obligatory command line option ``release.version``.
+To create next version marker use ``markNextVersion``. Without parameters next version will be created with default version incrementer - incrementMinor,
+witch can be overwritten with ``release.incrementer`` parameter.
```
Burn her, burn her!

pun-ky commented on src/main/groovy/pl/allegro/tech/build/axion/release/domain/NextVersionMarker.groovy:
```diff
@@ -9,22 +10,34 @@ class NextVersionMarker {
+        def version = nextVersionRules.nextVersion
+        def currentVersion = Version.valueOf(versionConfig.version)
+
+        if (nextVersionRules.versionIncrementer){
```
space after `)`

toniedzwiedz replied:
Formatting :)

pun-ky commented on src/main/groovy/pl/allegro/tech/build/axion/release/domain/PredefinedVersionIncrementer.groovy:
```diff
@@ -64,7 +64,7 @@ enum PredefinedVersionIncrementer {
         PredefinedVersionIncrementer creator = values().find { it.name == name }
         if (creator == null) {
             throw new IllegalArgumentException("There is no predefined version incrementer with $name name. " +
-                    "You can choose from: ${values().collect { it.name }}");
```
crucial changes ;) that semicolons...

pun-ky commented on src/main/groovy/pl/allegro/tech/build/axion/release/domain/scm/ScmService.groovy:
```diff
@@ -47,6 +47,10 @@ class ScmService {
         }
     }
 
+    ScmPosition position(){
```
space after `()`

toniedzwiedz replied:
Formatting

toniedzwiedz commented on docs/configuration/next_version.rst:
```diff
@@ -10,19 +10,21 @@ version number and last tag is next version marker, version is treated as SNAPSH
     2.0.0-SNAPSHOT
 
-To create next version marker use ``markNextVersion`` task along with obligatory command line option ``release.version``.
+To create next version marker use ``markNextVersion``. Without parameters next version will be created with default version incrementer - incrementMinor,
```
``incrementMinor``

?

jplucinski replied:
it's not Markdown, [https://pl.wikipedia.org/wiki/ReStructuredText](https://pl.wikipedia.org/wiki/ReStructuredText)

toniedzwiedz commented on src/main/groovy/pl/allegro/tech/build/axion/release/infrastructure/config/NextVersionPropertiesFactory.groovy:
```diff
@@ -8,16 +8,19 @@ class NextVersionPropertiesFactory {
         }
 
-        return new NextVersionProperties(nextVersion: nextVersion, suffix: config.suffix, separator: config.separator,
+        return new NextVersionProperties(nextVersion: nextVersion, versionIncrementer: nextVersionIncrementer, suffix: config.suffix, separator: config.separator,
```
This is technically a public constructor. How will it behave if we omit this parameter?

jplucinski replied:
I updated all references, moreover in project instances of NextVersionfactory are created with builder NextVersionPropertiesBuilder.

Commit:
feature/markNextVersionTask-creates-next-version-with-incrementer

* added examples in doc
* fixed formatting

lwasyl committed:
Fixes two commits on the same tag issue #236

adamdubiel committed:
Release version: 1.9.1

lwasyl committed:
Fix version resolution with multiple tags on history commits (issue #238)

jplucinski committed:
feature/markNextVersionTask-creates-next-version-with-incrementer

* using undecorated version for incrementer
* more explicit logic
* services clean up

jplucinski committed:
feature/markNextVersionTask-creates-next-version-with-incrementer

* move incrementer resolving to nextversionmaker
* fix for unit test

toniedzwiedz commented:
Jędrzej, idź spać.

adamdubiel committed:
Fixes Gradle integraiton test setup

Bug in integration tests setup code caused project.version
to be assigned before adding custom code. This means that
no version reading customizations were active in integraiton tests.

adamdubiel committed:
Moves version sorting to separate class

This enables better testing and isolation for
pretty sophisticated version sorting logic.

adamdubiel committed:
Fixes bugs with version sorting in different variants

Specifies precedence of tags when multiple tags on single commit:

* when on head: use highest normal; if no normal - first alpha
* otherwise: use highest of all

adamdubiel committed:
Release version: 1.9.2

adamdubiel committed:
Adds user-friendly message when no tracking branch set

Commit:
Merge branch 'master' into feature/markNextVersionTask-creates-next-version-with-incrementer

Commit:
feature/markNextVersionTask-creates-next-version-with-incrementer

*  remove VersionConfig dependencies from NextVersionPropertiesFactory
